### PR TITLE
fix: fix the problem that bot cannot be found.

### PR DESCRIPTION
**Pull request by ConstasJ**
根据我的测试，原有的代码会出现找到的bot对象为undefined这个问题。我在此基础上修改了这部分代码，测试结果合格。

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "koishi-plugin-forward",
+  "name": "@constasj/koishi-plugin-forward",
```
Comment (MaikoTan) [inline package.json:2]: You should NOT change the name of the plugin.

File: src/index.ts [hunk 2]
```diff
@@ -68,7 +68,12 @@ export function apply(ctx: Context, config: Config) {
         rule.guildId = channel.guildId
       }
 
-      const bot = ctx.bots[`${platform}:${rule.selfId}`]
+      // const bot = ctx.bots[`${platform}:${rule.selfId}`]
+      // According to my test result,this cannot work properly in the latest Koishi
+      // this is my solution
```
Comment (MaikoTan) [inline src/index.ts:73]: Reasonless comment, either remove it or change to a meaningful one.

File: .gitignore [hunk 3]
```diff
@@ -109,3 +109,4 @@ dist
 
 # TernJS port file
 .tern-port
+pnpm-lock.yaml
```
Comment (MaikoTan) [inline .gitignore:112]: Why pnpm here?

**Review by MaikoTan — CHANGES_REQUESTED** [review]
I don't think this change is correct.

It looks more likely that you have misconfigure the adapter plugin resulting a nil `selfId` causing that `ctx.bots[sid]` find nothing. I might be suspicion but it looks weird to change it with `.find()` function here.

File: package.json (hunk 1 above)
Comment (ConstasJ) [reply package.json]: I made a mistake.
Maybe I should checkout a new branch.

File: src/index.ts [hunk 4]
```diff
@@ -68,7 +68,9 @@ export function apply(ctx: Context, config: Config) {
         rule.guildId = channel.guildId
       }
 
-      const bot = ctx.bots[`${platform}:${rule.selfId}`]
+      const bot = ctx.bots.find((bot)=>{
+        return bot.platform === platform && (bot.userId === rule.selfId || bot.selfId === rule.selfId )
```
Comment (shigma) [inline src/index.ts:72]: 这个应该是 bot 实现有问题，并非这个插件的问题。

**Pull request closed without merging**